pkg/splunk/enterprise/util.go

```diff
@@ -48,3 +52,64 @@ func ApplySplunkConfig(client splcommon.ControllerClient, cr splcommon.MetaObjec
 
 	return namespaceScopedSecret, nil
 }
+
+// GetSmartstoreSecrets is used to retrieve S3 access key and secrete keys.
+// To do: sgontla:
+// 1. Support multiple secret objects
+// 2. default secret object
+// volume name other than default, look for the specific secret object, else fetch from defaults
+func GetSmartstoreSecrets(client splcommon.ControllerClient, cr splcommon.MetaObject, smartstore *enterprisev1.SmartStoreSpec) (string, string, error) {
+	var defaultSplunksecret corev1.Secret
+
+	namespacedName := types.NamespacedName{
+		Namespace: cr.GetNamespace(),
+		Name:      GetNamespaceScopedSecretName(cr.GetNamespace()),
+	}
+	err := client.Get(context.TODO(), namespacedName, &defaultSplunksecret)
+	if err != nil {
+		return "", "", err
+	}
+
+	return string(defaultSplunksecret.Data[s3AccessKey]), string(defaultSplunksecret.Data[s3SecretKey]), nil
```

Comment:
What if there are no values for keys s3AccessKey and s3SecretKey? Should we add a preventive check before?